fastchat/model/compression.py

```diff
@@ -117,15 +129,36 @@ def load_compress_model(model_path, device, torch_dtype, use_fast, revision="mai
             model_path,
             low_cpu_mem_usage=True,
             torch_dtype=torch_dtype,
-            trust_remote_code=True,
             revision=revision,
+            trust_remote_code=True,
         )
         # some models are loaded by AutoModel but not AutoModelForCausalLM,
         # such as chatglm, chatglm2
         try:
             model = AutoModelForCausalLM.from_config(config, trust_remote_code=True)
         except NameError:
             model = AutoModel.from_config(config, trust_remote_code=True)
+        if not max_gpu_memory:
+            max_memory = get_balanced_memory(
+                model,
+                dtype=torch_dtype,
+                low_zero=False,
+                no_split_module_classes=model._no_split_modules,
+            )
+        else:
+            import re
+            import psutil
+
+            gbit = int(float(re.search("[0-9]+", max_gpu_memory).group(0)) * 1024**3)
+            max_memory = {i: gbit for i in range(num_gpus)}
+            max_memory["cpu"] = psutil.virtual_memory().available
```

Comment:
Try to not enable cpu offload by default and remove the dependency of psutil

Reply:
cpu-offload would not be executed if there is enough gpu memory. I think accelerate.infer_auto_device_map can handle that very well.